samtranslator/intrinsics/actions.py

```diff
@@ -490,6 +490,12 @@ def resolve_resource_id_refs(self, input_dict, supported_resource_id_refs):
         if not isinstance(value, list) or len(value) < 2:
             return input_dict
 
+        # If items in value array is not a string, then following join line will fail. So if any element is not a string
+        # we just pass along the input to CFN for doing the validation
+        for item in value:
+            if not isinstance(item, string_types):
+                return input_dict
```

Comment:
We may also want to update line 430.
There, we error out if one of the values is not a string.

Based on the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html) (Supported Function, at the bottom of the page), it seems that the resource name must be a string but the attribute name can be a `Ref`.

We may also want to add a test in `tests/translator/input/intrinsic_functions.yaml` with this new case we allow.